compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1338,7 +1339,7 @@ impl<'tcx> TyCtxt<'tcx> {
 
     pub fn serialize_query_result_cache<E>(self, encoder: &mut E) -> Result<(), E::Error>
     where
-        E: ty::codec::OpaqueEncoder,
+        E: OpaqueEncoder,
```

Comment:
This function is only ever called with `FileEncoder`. It could be made monomorphic.

Reply:
`OnDiskCache::serialize` and `encode_query_results` could also.